packages/wrangler/templates/pages-template-worker.ts

```diff
@@ -126,19 +126,20 @@ export default {
           waitUntil: workerContext.waitUntil.bind(workerContext),
         };
 
-        const response = await handler(context);
+        const response: Response = await handler(context);
```

Comment:
What if `handler` doesn't return an instance of `Response`? This is effectively just coercing an `any` to `Response`, which we should be pretty careful about. Maybe we can remove the type annotation and then perform an `instanceof` check instead?

```tsx
const response = await handler(context);

if (!(response instanceof Response)) {
  throw new Error('...')
}
```

Reply:
Good thinking. I think I'm fairly certain we require handlers to return an instance of `Response` though?

Reply:
AFAICT the `handler` is already of type: `PagesFunction<unknown, string, Record<string, unknown>>`, which has the following call-member:

```ts
(context: EventContext<Env, P, Data>) => Response | Promise<Response>
```

So calling `await handler()` should already be a `Response` not `any`. So there should be no need for the `: Response` type here and no need to do any casting.